### PR TITLE
Support additional properties

### PR DESCRIPTION
**Pull request by pakrym**
I'll add tests.

Move more logic into `JsonSerializationExtensions`

### CODE REVIEW

File: src/assets/Azure.Core.Shared/JsonElementExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    internal static class JsonElementExtensions
+    {
+        public static object? GetObject(in this JsonElement element)
+        {
+            switch (element.ValueKind)
+            {
+                case JsonValueKind.String:
+                    return element.GetString();
+                case JsonValueKind.Number:
+                    return element.GetInt32();
```
Comment (MiYanni) [inline src/assets/Azure.Core.Shared/JsonElementExtensions.cs]: Shouldn't we assume the largest amount of possible numeric types, and use decimal? (not int)

File: test/TestServerProjects/body-boolean/Generated/Models/Error.Serialization.cs [hunk 2]
```diff
@@ -20,6 +21,7 @@ internal static void Serialize(Error model, Utf8JsonWriter writer)
                 writer.WritePropertyName("message");
                 writer.WriteStringValue(model.Message);
             }
+
```
Comment (MiYanni) [inline test/TestServerProjects/body-boolean/Generated/Models/Error.Serialization.cs]: It seems there is an extra carriage return added to both the Serialize and Deserialize methods.

File: src/AutoRest.CSharp.V3/ClientModels/ModelBuilder.cs [hunk 3]
```diff
@@ -25,11 +25,28 @@ internal class ModelBuilder
 
         private static ClientModel BuildClientObject(ObjectSchema objectSchema)
         {
-            var inheritsFrom = objectSchema.Parents?.Immediate.OfType<ObjectSchema>().SingleOrDefault();
-            var inheritsFromTypeReference = inheritsFrom != null ? ClientModelBuilderHelpers.CreateType(inheritsFrom, false) : null;
+            ClientTypeReference? inheritsFromTypeReference = null;
+            DictionaryTypeReference? dictionaryElementTypeReference = null;
+
+            foreach (ComplexSchema complexSchema in objectSchema.Parents!.Immediate)
```
[the remark this answers is not in the document]
Comment (pakrym) [reply src/AutoRest.CSharp.V3/ClientModels/ModelBuilder.cs]: Yep, https://github.com/Azure/autorest.csharp/pull/358/files#diff-5b490df2c1adda7893a3eddd9241e459R21

File: src/assets/Azure.Core.Shared/JsonElementExtensions.cs (hunk 1 above)
Comment (pakrym) [reply src/assets/Azure.Core.Shared/JsonElementExtensions.cs]: Decimal seems overkill, I did double.

**Pull request merged**